### ISSUE TITLE
[iOS, Android] Keyframe animation of Rotation is not handled correctly

### ISSUE THREAD

**Issue by BrianDT** [unoplatform/uno#12788]
### Current behavior

On iOS the animated object will be translated off screen at the end of the animation if the angle is not set to zero.

On iOS the animation of rotation at the same time as translation does not require the animation of the CenterX and CenterY properties which is required on Windows.
If the animation of the CenterX and CenterY properties are added on iOS then the storyboard completion event is never fired.

Works correctly on Windows and WASM.

Android has different problems which have been covered in other issues that need to be resolved however there is also a rotation problem in that including the animation of the CenterX and CenterY properties causes the target to be animated incorrectly.

On Android the storyboard completion event is fired however the target moves back to its original location which is incorrect.

### Expected behavior

The animation of Rotation on iOS should work the same as it does on Windows, code should not need to be removed for the animation to complete. 

### How to reproduce it (as minimally and precisely as possible)

Reproduced in the sample at 
https://github.com/BrianDT/UnoAnimationCompleteIssue

In the sample the code to add the unnecessary final change to the rotation and to remove the animation of CentreX and CentreY is included using ifdefs. These can be adjusted to show the undesirable effects.

### Workaround

The inclusion of the final rotation to zero and the removal of the CenterX and CenterY animations on iOS bypass some of the issues. However, the inclusion of the unnecessary final change to the rotation can lead to inappropriate changes in the UI of the app which mean object positions are out of context.

### Works on UWP/WinUI

Yes

### Environment

Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

4.9.26

### Affected platforms

Android, iOS

### IDE

Visual Studio 2019

### IDE version

17.6.4

**Comment by jeromelaban** [comment]
@dr1rrb does this look familiar?

**Comment by dr1rrb** [comment]
Yes, you are animating 2 properties of the same object. This is one of the known limitation on iOS and Android https://github.com/unoplatform/uno/issues/12530 (https://github.com/BrianDT/UnoAnimationCompleteIssue/blob/df36acfed58eec0dd35df11ff083aa18ab50f22e/UnoAnimationCompleteIssue/MainPage.xaml.cs#L194 at line 194, 204 and 214).

This is because on those platform each animation is delegated to the OS but each is run independently. At the end we apply the "final transformation" no matter if some other animations are still running.

FYI Here is the epic that track some other issues regarding animations: https://github.com/unoplatform/uno/issues/3919

**Comment by BrianDT** [comment]
Sample updated to 5.0.40, confirm still an issue.

**Comment by BrianDT** [comment]
Sample updated to 5.2.175, confirm still an issue.

**Comment by MartinZikmund** [comment]
Only relevant to native targets, should work correctly on Skia renderer

**Comment by BrianDT** [comment]
> Only relevant to native targets, should work correctly on Skia renderer

Sample updated, confirm resolved
